projects/hallucination/README.md

```diff
@@ -45,8 +45,95 @@ To use the modified splits as described in the [paper](https://arxiv.org/abs/210
+    -lr 1e-05 -vmm min -veps 0.25 -vme 1000 -vmt ppl -vp 5 \
+
+#### Train a T5-Large RAG-Turn Doc-Then-Turn model with DPR Retrieval on WoW
+
+    parlai train_model -m rag -t wizard_of_wikipedia \
+    --rag-model-type turn --rag-turn-marginalize doc_then_turn --rag-retriever-type dpr \
+    --generation-model t5 --t5-model-arch t5-large \
+    --batchsize 16 --fp16 True --gradient-clip 0.1 --label-truncate 128 \
+    --log-every-n-secs 30 --lr-scheduler reduceonplateau --lr-scheduler-patience 1 \
+    --model-parallel True --optimizer adam --text-truncate 512 --truncate 512 \
+    -lr 1e-05 -vmm min -veps 0.25 -vme 1000 -vmt ppl -vp 5 \
+
+#### Train a BlenderBot-2.7B RAG Sequence Model with DPR-Poly Retrieval on WoW
+
+For the BlenderBot model, we add extra positions to the encoder, so that we can retain additional information from the retrieved documents.
+
+    parlai train_model -m rag -t wizard_of_wikipedia \
+    --rag-model-type turn --rag-turn-marginalize doc_then_turn --rag-retriever-type dpr \
+    --generation-model transformer/generator -o arch/blenderbot_3B \
+    --n-extra-positions 128 \
+    --init-model zoo:blender/blender_3B/model --dict-file zoo:blender/blender_3B/model.dict \
+    --batchsize 8 --fp16 True --gradient-clip 0.1 \
+    --log-every-n-secs 30 --lr-scheduler reduceonplateau --lr-scheduler-patience 1 \
+    --model-parallel True --optimizer adam \
+    -lr 1e-05 -vmm min -veps 0.25 -vme 1000 -vmt ppl -vp 5 \
+
+#### Train a BART-Large FiD Model, with a DPR Retriever initialized from a DPR Model trained with RAG.
+
+This is the **BART FiD RAG** model specified above.
+
+    parlai train_model -m fid -t wizard_of_wikipedia \
+    --rag-retriever-type dpr --query-model bert_from_parlai_rag \
+    --dpr-model-file zoo:hallucination/bart_rag_token/model \
+    --generation-model bart -o arch/bart_large \
+    --batchsize 16 --fp16 True --gradient-clip 0.1 --label-truncate 128 \
+    --log-every-n-secs 30 --lr-scheduler reduceonplateau --lr-scheduler-patience 1 \
+    --model-parallel True --optimizer adam --text-truncate 512 --truncate 512 \
+    -lr 1e-05 -vmm min -veps 0.25 -vme 1000 -vmt ppl -vp 5 \
+
+#### Train a T5-Base FiD Model, using a smaller index for debug purposes.
+
+We provide a smaller FAISS index comprising ~3k documents, which encompasses all topics appearing in the Wizard of Wikipedia dataset.
+
+    parlai train_model -m fid -t wizard_of_wikipedia \
+    --rag-retriever-type dpr --query-model bert_from_parlai_rag \
+    --dpr-model-file zoo:hallucination/bart_rag_token/model \
+    --retriever-small-index exact \
+    --generation-model t5 --t5-model-arch t5-base \
+    --batchsize 16 --fp16 True --gradient-clip 0.1 --label-truncate 128 \
+    --log-every-n-secs 30 --lr-scheduler reduceonplateau --lr-scheduler-patience 1 \
+    --model-parallel True --optimizer adam --text-truncate 512 --truncate 512 \
+    -lr 1e-05 -vmm min -veps 0.25 -vme 1000 -vmt ppl -vp 5 \
```

Comment:
maybe add a note here to open an issue if other options are desired?

Reply:
Option prefixes are available now...

Reply:
i know i added one for BART. i wasnt sure what to call these though... `opt/rag`? since these are optimization/training parameters

Reply:
you can put them in the project folder and specify it with `-o projects/hallucination/very_long_name.opt`

Reply:
wow, wasn't aware of that, i'll try it out